spark-commons/src/main/scala/za/co/absa/spark/commons/implicits/DataFrameImplicits.scala

```diff
@@ -93,6 +99,22 @@ object DataFrameImplicits {
      * @return Returns aligned and filtered utils
      */
     def alignSchema(structType: StructType): DataFrame = alignSchema(structType.getDataFrameSelector())
+
+    /**
+     * Persist this Dataset with the default storage level (`MEMORY_AND_DISK`), the same way as
+     * [[org.apache.spark.sql.Dataset.cache]] does. But does not throw a warning if the DataFrame has been cached before.
+     *
+     * @return the DataFrame itself
+     * @since 0.3.0
+     */
+    def cacheIfNot(): DataFrame = {
+      val planToCache = df.queryExecution.analyzed
+      if (df.sparkSession.sharedState.cacheManager.lookupCachedData(planToCache).isEmpty) {
+        df.cache()
```

Comment:
I just had a look at this in `spark-partition-sizing` , since some tests were failing on the original implementation and I think it may need to be:

```suggestion
        df.cache(_=>())
```

Reply:
`cache` doesn't take parameters. I do remember using the identity somewhere (of partition resizing). But not directly in `cache`.